test/vfd_swmr_gfail_writer.c

```diff
@@ -0,0 +1,798 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of HDF5.  The full HDF5 copyright notice, including     *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the root of the source code       *
+ * distribution tree, or in https://www.hdfgroup.org/licenses.               *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*
+ *  Purpose: To demostrate the four issues uncovered during the test plan
+ *           1.4 part 1.
+ *
+ *           Setup:
+ *           Writer:
+ *            1) Create an HDF5 file
+ *            2) Create many groups
+ *            3) Use the named pipe to send the reader a message to start verifying
+ *            4) Call H5Fvfd_swmr_end_tick immeidately after sending out the message to the reader
+ *            5) Sleep for two ticks before receiving a message from the reader that informs the reader
+ *               started verifying
+ *            6) If the option to delete the group is off(by default), just sleep for a relatively long time,
+ *                  then close the HDF5 file.
+ *               Else delete the last 1000 groups that are just created if the total number of
+ *                  created groups is greater than 1000.
+ *                  Call H5Fvfd_swmr_end_tick sleep for a relatively long time,
+ *                  then close HDF5 file.
+ *          Reader:
+ *            1) Open the HDF5 file
+ *            2) Wait for the writer's message that informs the writer finished creating groups.
+ *            3) After receiving the message from the writer, send a message back to the writer,
+ *               then call H5Literate to iterate through all the groups. The callback function just checks if
+ *               the group name's prefix is valid.
+ *               An #if 0 #endif block can help the user easily tune to check the iterated group names.
+ *            4) HDclock_gettime is used to check the total time of H5Literate call.
+ *            5) Close the HDF5 file.
+ *
+ *          The number of groups, the tick length, the max lag, the page buffer size, the page size and the
+ *          sleep duration on the writer side before closing the file are configurable.
+ *          Users can also choose an option to delete 1000 groups after creating a larger number of groups..
```

Comment:
trivial spelling correction:
...creating a larger number of groups.
We only test to create the groups with the latest file format.....
group via the earliest file format is still there.